### PR TITLE
Prune deprecated trainer attributes

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

remove deprecated trainer attributes such as `use_*` or `on_*`

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/accelerators/test_accelerator_connector.py [hunk 1]
```diff
@@ -105,7 +106,7 @@ def test_accelerator_choice_ddp_slurm(setup_distributed_mock):
     class CB(Callback):
 
         def on_fit_start(self, trainer, pl_module):
-            assert trainer.use_ddp
+            assert trainer.accelerator_connector._distrib_type in (DistributedType.DDP, DistributedType.DDP_SPAWN)
```
Comment (awaelchli) [inline tests/accelerators/test_accelerator_connector.py:109]: I think we can remove these lines completely.
We already have the check `assert isinstance(trainer.training_type_plugin, DDPPlugin)` below.

**Review by carmocca — APPROVED** [review]

File: tests/accelerators/test_accelerator_connector.py (hunk 1 above)
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:109]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 2]
```diff
@@ -144,7 +145,7 @@ def test_accelerator_choice_ddp2_slurm(device_count_mock, setup_distributed_mock
     class CB(Callback):
 
         def on_fit_start(self, trainer, pl_module):
-            assert trainer.use_ddp2
+            assert trainer.accelerator_connector._distrib_type == DistributedType.DDP2
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:148]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 3]
```diff
@@ -183,7 +184,7 @@ def test_accelerator_choice_ddp_te(device_count_mock, setup_distributed_mock):
     class CB(Callback):
 
         def on_fit_start(self, trainer, pl_module):
-            assert trainer.use_ddp
+            assert trainer.accelerator_connector._distrib_type in (DistributedType.DDP, DistributedType.DDP_SPAWN)
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:187]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 4]
```diff
@@ -221,7 +222,7 @@ def test_accelerator_choice_ddp2_te(device_count_mock, setup_distributed_mock):
     class CB(Callback):
 
         def on_fit_start(self, trainer, pl_module):
-            assert trainer.use_ddp2
+            assert trainer.accelerator_connector._distrib_type == DistributedType.DDP2
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:225]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 5]
```diff
@@ -257,7 +258,7 @@ def test_accelerator_choice_ddp_cpu_te(device_count_mock, setup_distributed_mock
     class CB(Callback):
 
         def on_fit_start(self, trainer, pl_module):
-            assert trainer.use_ddp
+            assert trainer.accelerator_connector._distrib_type in (DistributedType.DDP, DistributedType.DDP_SPAWN)
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:261]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 6]
```diff
@@ -294,7 +295,7 @@ def test_accelerator_choice_ddp_cpu_slurm(device_count_mock, setup_distributed_m
...
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:298]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 7]
```diff
@@ -342,7 +343,7 @@ def creates_children(self) -> bool:
...
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:346]:
```suggestion
```

File: tests/accelerators/test_accelerator_connector.py [hunk 8]
```diff
@@ -36,6 +36,7 @@
     SingleDevicePlugin,
 )
 from pytorch_lightning.plugins.environments import LightningEnvironment, SLURMEnvironment, TorchElasticEnvironment
+from pytorch_lightning.utilities import DistributedType
```
Comment (Borda) [inline tests/accelerators/test_accelerator_connector.py:39]:
```suggestion
```

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by ananthsub — DISMISSED** [review]
more tests need to be cleaned up to reflect the removed properties

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Pull request merged**